db/migrate/20150707214004_add_root_approvals.rb

```diff
@@ -0,0 +1,28 @@
+class AddRootApprovals < ActiveRecord::Migration
+  def up
+    execute <<-SQL
+      INSERT INTO approvals (status, position, proposal_id, type)
+      SELECT 'actionable', -1, proposals.id, 'Approvals::Parallel'
+      FROM proposals WHERE proposals.flow = 'parallel';
+    SQL
+    execute <<-SQL
+      INSERT INTO approvals (status, position, proposal_id, type)
+      SELECT 'actionable', -1, proposals.id, 'Approvals::Serial'
+      FROM proposals WHERE proposals.flow = 'linear';
+    SQL
+    execute <<-SQL
+      UPDATE approvals
+      SET parent_id = (SELECT root.id
+                       FROM approvals root
+                       WHERE root.proposal_id = approvals.proposal_id
+                       AND root.type <> 'Approvals::Individual')
+      WHERE type = 'Approvals::Individual';
+    SQL
+  end
```

Comment:
This migration seems complex enough that we may want to run it against a snapshot of production data...thoughts?